src/test/regress/input/multi_copy.source

```diff
@@ -781,6 +781,17 @@ SELECT create_distributed_table('tt1','id');
 DROP TABLE tt1;
 END;
 
+-- Test dropping a column in front of the partition column
```

Comment:
I can see that the code doesn't deal with the columns after the partition column. However, to guard against any future issue, shall we add a test after dropping `col4` as well?